src/redis_disk.cc

```diff
@@ -0,0 +1,151 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#include <vector>
+#include <memory>
+#include <utility>
+#include <algorithm>
+#include <string>
+
+#include "db_util.h"
+#include "redis_metadata.h"
+#include "redis_sortedint.h"
+#include "rocksdb/status.h"
+#include "redis_zset.h"
+#include "redis_bitmap.h"
+#include "redis_disk.h"
+#include "status.h"
+
+namespace Redis {
+rocksdb::Status Disk::GetApproximateSizes(const Metadata &metadata, const Slice &ns_key,
+                                          rocksdb::ColumnFamilyHandle *column_family,
+                                          uint64_t *key_size, Slice subkeyleft,
+                                          Slice subkeyright) {
+  std::string prefix_key, next_version_prefix_key;
+  InternalKey(ns_key, subkeyleft, metadata.version,
+              storage_->IsSlotIdEncoded()).Encode(&prefix_key);
+  InternalKey(ns_key, subkeyright, metadata.version + 1,
+              storage_->IsSlotIdEncoded()).Encode(&next_version_prefix_key);
+  auto key_range = rocksdb::Range(prefix_key, next_version_prefix_key);
+  uint64_t tmp_size = 0;
+  rocksdb::Status s = db_->GetApproximateSizes(option_, column_family,
+                                               &key_range, 1, &tmp_size);
+  if (!s.ok()) return s;
+  *key_size += tmp_size;
+  return rocksdb::Status::OK();
+}
+rocksdb::Status Disk::GetKeySize(const Slice &user_key, RedisType type, uint64_t *key_size) {
+  *key_size = 0;
+  std::string ns_key;
+  AppendNamespacePrefix(user_key, &ns_key);
+  switch (type) {
+    case RedisType::kRedisString:
+      return GetStringSize(ns_key, key_size);
+    case RedisType::kRedisHash:
+      return GetHashSize(ns_key, key_size);
+    case RedisType::kRedisBitmap:
+      return GetBitmapSize(ns_key, key_size);
+    case RedisType::kRedisList:
+      return GetListSize(ns_key, key_size);
+    case RedisType::kRedisSet:
+      return GetSetSize(ns_key, key_size);
+    case RedisType::kRedisSortedint:
+      return GetSortedintSize(ns_key, key_size);
+    case RedisType::kRedisZSet:
+      return GetZsetSize(ns_key, key_size);
+    case RedisType::kRedisNone:
+      return rocksdb::Status::NotFound("Not found ", user_key);
+    case RedisType::kRedisStream:
+      return rocksdb::Status::NotSupported("Not support stream");
```

Comment:
Why Stream type is not supported? I think getting its size is very similar to `set`, except Stream has its own column family. By the way, you can add a new line before and after `GetKeySize` and delete one line between `GetHashSize` and `GetSetSize`.

Reply:
@torwig  I'm sorry for missing the `stream`. I added the corresponding type based on [design document](https://kvrocks.apache.org/docs/Design/design-structure-on-rocksdb). There is no documentation for `stream` here. I noticed that `stream` are implemented by you, do you have time to add `stream`  into [this website](https://kvrocks.apache.org/docs/Design/design-structure-on-rocksdb)?

Reply:
Yes, I can add the design of the Stream data type to that document. I just thought that that document can be updated with the changes that are released. And right now the Stream type isn't released. @git-hulk @PragmaTwice Am I right?

Reply:
Sure, I had updated the supported command in the Kvrocks website: https://github.com/apache/incubator-kvrocks-website/blob/main/docs/01-supported-commands.md

Reply:
I see, then I will update the design docs.

Reply:
Thanks @torwig

Reply:
@tanruixiang I've added Stream design here https://github.com/apache/incubator-kvrocks-website/pull/17 .

Reply:
> @tanruixiang I've added Stream design here [apache/incubator-kvrocks-website#17](https://github.com/apache/incubator-kvrocks-website/pull/17) .

Thank you very much! Thanks for your contribution!

Reply:
I'm still looking for the reason why sometimes the value obtained is 0. If found, I will add  `GetStreamSize`.